From 427dab431f7e8c4c8a01e9e9ed0892708a3d22d2 Mon Sep 17 00:00:00 2001
From: Atin Mukherjee <amukherj@redhat.com>
Date: Mon, 5 Aug 2019 08:33:29 +0530
Subject: [PATCH 263/265] tests: fix ctime related tests

With ctime being disabled by default, certain tests need to explicitly
turn this option off to sanitize the functionality

Label: DOWNSTREAM ONLY

Change-Id: Id70310b4b09e36bf66756fea447186bb073b5604
BUG: 1704562
Signed-off-by: Atin Mukherjee <amukherj@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/177704
Tested-by: RHGS Build Bot <nigelb@redhat.com>
Reviewed-by: Sunil Kumar Heggodu Gopala Acharya <sheggodu@redhat.com>
---
 tests/basic/ctime/ctime-noatime.t | 1 +
 tests/basic/ctime/ctime-readdir.t | 1 +
 tests/bugs/glusterd/bug-1696046.t | 8 +++++---
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/tests/basic/ctime/ctime-noatime.t b/tests/basic/ctime/ctime-noatime.t
index 609ccbd..a6c8d9c 100644
--- a/tests/basic/ctime/ctime-noatime.t
+++ b/tests/basic/ctime/ctime-noatime.t
@@ -20,6 +20,7 @@ function atime_compare {
 TEST glusterd
 TEST pidof glusterd
 TEST $CLI volume create $V0 replica 3 $H0:$B0/${V0}{0,1,2}
+TEST $CLI volume set $V0 ctime on
 TEST $CLI volume set $V0 performance.stat-prefetch off
 TEST $CLI volume set $V0 performance.read-ahead off
 TEST $CLI volume set $V0 performance.quick-read off
diff --git a/tests/basic/ctime/ctime-readdir.t b/tests/basic/ctime/ctime-readdir.t
index 4564fc1..fa069b3 100644
--- a/tests/basic/ctime/ctime-readdir.t
+++ b/tests/basic/ctime/ctime-readdir.t
@@ -9,6 +9,7 @@ TEST glusterd
 
 TEST $CLI volume create $V0 replica 3 ${H0}:$B0/brick{1,2,3};
 TEST $CLI volume set $V0 performance.stat-prefetch on
+TEST $CLI volume set $V0 ctime on
 TEST $CLI volume set $V0 performance.readdir-ahead off
 TEST $CLI volume start $V0;
 
diff --git a/tests/bugs/glusterd/bug-1696046.t b/tests/bugs/glusterd/bug-1696046.t
index e1c1eb2..f7992f5 100644
--- a/tests/bugs/glusterd/bug-1696046.t
+++ b/tests/bugs/glusterd/bug-1696046.t
@@ -22,6 +22,8 @@ TEST pidof glusterd;
 TEST $CLI volume set all cluster.brick-multiplex on
 TEST $CLI volume create $V0 replica 3 $H0:$B0/${V0}{1,2,3};
 TEST $CLI volume create $V1 replica 3 $H0:$B0/${V1}{1,2,3};
+TEST $CLI volume set $V0 ctime on
+TEST $CLI volume set $V1 ctime on
 
 ## Start volume and verify
 TEST $CLI volume start $V0;
@@ -64,9 +66,9 @@ TEST $CLI volume set $V0 diagnostics.brick-log-level DEBUG
 # Do some operation
 touch $M0/file1
 
-# Check debug message debug message should be exist only for V0
-# Server xlator is common in brick_mux so after enabling DEBUG log
-# some debug message should be available for other xlators like posix
+# Check debug message should exist only for V0 server xlator is common in
+# brick_mux so after enabling DEBUG log some debug message should be available
+# for other xlators like posix
 
 brick_log_file=$logdir"/bricks/"`brick-log-file-name`
 nofdlog=$(cat $brick_log_file | grep file1 | grep -v server | wc -l)
-- 
1.8.3.1

